src/callsystems/legacy/index.js

```diff
@@ -0,0 +1,80 @@
+import { ModelInteractions } from "../../util/models/index.js";
+import { Callsystem } from "../../lib/callsystems/index.js";
+
+export default class Legacy extends Callsystem {
+	constructor(opts) {
+		const { env, message, client, defaultModel, defaultProvider } = opts || {};
+		super({ env, message, client, defaultModel, defaultProvider });
+	}
+
+	static get packageId() {
+		return "cs.spongedsc.legacy";
+	}
+
+	static get name() {
+		return "Legacy";
+	}
+
+	static get version() {
+		return "2.0.1";
+	}
+
+	static get capabilities() {
+		return ["legacy"];
+	}
+
+	async activate() {
+		const { message, client, env } = this;
+		const modelInteractions = new ModelInteractions({
+			message,
+			kv: client.kv,
+			instructionSet: client.tempStore.get("instructionSet") || env.MODEL_LLM_PRESET || "default",
+			baseHistory: [],
+			accountId: env.CLOUDFLARE_ACCOUNT_ID,
+			token: env.CLOUDFLARE_ACCOUNT_TOKEN,
+			model: "@cf/meta/llama-3-8b-instruct",
+		});
+
+		const formattedMessage = await modelInteractions.response.formatUserMessage();
+		const validityCheck = await modelInteractions.messageEvent.validateHistory();
+
+		if (!validityCheck.valid) {
+			if (validityCheck?.handled?.isRequired && !validityCheck?.handled?.executed)
+				return await message.react("⚠️").catch(() => {
+					console.log("Failed to delete key from KV after valid history check failed");
+					console.error(e);
+					return false;
+				});
+		}
+
+		const history = await modelInteractions.history
+			.add({
+				key: message?.channel?.id,
+				role: "user",
```

Comment:
ok i know we already do this but why not use a system role instead?

Reply:
i just moved everything over from the old legacy backend, but the 8B variant of LLaMA doesn't have a very good context window